pkg/disk-cache/dclient.go

```diff
@@ -0,0 +1,81 @@
+package diskcache
+
+import (
+	"context"
+	pb "github.com/anujga/dstk/pkg/api/proto"
+	"github.com/anujga/dstk/pkg/core"
+	se "github.com/anujga/dstk/pkg/sharding_engine"
+	"google.golang.org/grpc"
+	"time"
+)
+
+type Client interface {
+	Get(key core.KeyT) ([]byte, error)
+	Put(key, value core.KeyT) error
+	Remove(key core.KeyT) error
+}
+
+type impl struct {
+	tc     se.ThickClient
+	rpcMap *core.ConcurrentMap
```

Comment:
This is already implemented and called connection pool

Reply:
reused the one in core